### DISCUSSION TITLE
Is there a difference between create and createMany when used in nested writes?

### DISCUSSION THREAD

**Discussion by ymc9** [prisma/prisma#15898] [Q&A] [answered]
When using nested write to create multiple nested entities, it seems I can use both create and createMany to achieve the same goal. Is there any difference between them?

**Comment by Rackar** [comment]
create is for one record.

**Reply by miller-productions** [reply]
When you say 
> create is for one record

I'm not sure this is exactly the case, I believe (_correct me if I'm wrong_) you can produce multiple rows using a `create` by passing it an array of items, e.g. `create: [{item: "one"}, {item: "two"}]`

**Comment by nurul3101** [comment]
Hello @ymc9 👋 

[createMany](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#createmany) would allow you to simultaneously insert multiple records in one go, while [create](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#create) would only allow creating one nested record. Nested writes are also done in a transaction so there would be no other significant difference.
Let me know if you have any other queries.

**Comment by ymc9** [comment, marked as answer]
Thank you, guys. I just noticed Prisma has nice documentation explaining the differences, and I missed it before raising this discussion:

https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-single-record-and-multiple-related-records

**Comment by miller-productions** [comment]
I've been thinking the same thing myself recently..

i.e. for example, you could either...

```
createMany: {
    data: [{title: "one"}, {title: "two} ],
}
```

or you could

```
create: [
    {title: "one"}, 
    {title: "two"},
]
```
Both end up giving you the same result.

<_note: I've chopped a whole lot from the code examples here just for brevity, see the full example in [the docs](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-single-record-and-multiple-related-records)_>

I think the key difference is described in [the docs](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-single-record-and-multiple-related-records) where they say

> Note that it is not possible to nest an additional create or createMany inside the highlighted query [i.e. inside a `createMany`]

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
